### PR TITLE
:sparkles: Adding ScheduleEvent

### PR DESCRIPTION
**Pull request by Sigmanificient**
### Changes

-   `adds`: ScheduleEvent

 ### Check off the following

-   [x] I have tested my changes with the current requirements
-   [x] My Code follows the pep8 code style.

### CODE REVIEW

File: pincer/objects/guild/schedule_events.py [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+    ----------
+    id: :class:`int`
+        The ID of the scheduled event.
+    name: :class:`str`
+        The name of the scheduled event (1-100 characters)
+    guild_id: :class:`int`
+        The guild id which the scheduled event belongs to
+    scheduled_start_time: str
+        The scheduled start time of the event.
+    privacy_level: :class:`PrivacyLevel`
```
Comment (Lunarmagpie) [inline pincer/objects/guild/schedule_events.py:65]:
```suggestion
    privacy_level: :class:`~pincer.guild.stage.PrivacyLevel`
```

File: pincer/objects/guild/schedule_events.py [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+        The privacy level of the scheduled event.
+    status: :class:`EventStatus`
```
Comment (Lunarmagpie) [inline pincer/objects/guild/schedule_events.py:67]:
```suggestion
    status: :class:`~pincer.guild.schedule_events.EventStatus`
```

File: pincer/objects/guild/schedule_events.py [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+        The status of the scheduled event.
+    entity_type: :class:`GuildScheduledEventEntityType`
```
Comment (Lunarmagpie) [inline pincer/objects/guild/schedule_events.py:69]:
```suggestion
    entity_type: :class:`~pincer.guild.schedule_events.GuildScheduledEventEntityType`
```

File: pincer/objects/guild/schedule_events.py [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+        The user id of the creator of the scheduled event
+    scheduled_end_time: str
+        The time the scheduled event will end, required if entity_type is EXTERNAL
+    description: :class:`str`
+        The description of the scheduled event (0-1000 characters)
+    entity_id: :class:`int`
+        The id of an entity associated with a guild scheduled event
+    entity_metadata: :class:`str`
+        Additional metadata for the guild scheduled event
+    creator: :class:`User`
```
Comment (Lunarmagpie) [inline pincer/objects/guild/schedule_events.py:84]:
```suggestion
    creator: :class:`~pincer.objects.user.user.User`
```

**Review by Lunarmagpie — CHANGES_REQUESTED** [review]
I think this file should be renamed to `scheduled_events.py`

**Review by nerdguyahmad — CHANGES_REQUESTED** [review]

**Review by Lunarmagpie — APPROVED** [review]

**Review by VincentRPS — APPROVED** [review]

File: pincer/objects/guild/scheduled_events.py [hunk 5]
```diff
@@ -0,0 +1,107 @@
...
+    COMPLETED: int
+        The event is completed.
+    """
+    SCHEDULED = 1
+    ACTIVE = 2
+    COMPLETED = 3
+    CANCELLED = 4
+
+
+class GuildScheduledEventEntityType(Enum):
```
Comment (trag1c) [inline pincer/objects/guild/scheduled_events.py:34]:
```suggestion
class GuildScheduledEventEntityType(IntEnum):
```

File: pincer/objects/guild/scheduled_events.py [hunk 6]
```diff
@@ -0,0 +1,107 @@
...
+    creator: :class:`~pincer.objects.user.user.User`
+        The user who created the scheduled event
+    user_count: :class:`int`
+        The number of users who have joined the scheduled event
+    """
+    id: Snowflake
+    name: str
+    guild_id: Snowflake
+    scheduled_start_time: Timestamp
+    privacy_level = PrivacyLevel
```
Comment (trag1c) [inline pincer/objects/guild/scheduled_events.py:95]:
```suggestion
    privacy_level: PrivacyLevel
```

File: pincer/objects/guild/guild.py [hunk 7]
```diff
@@ -3,23 +3,21 @@
...
 from enum import IntEnum
 from typing import AsyncGenerator, overload, TYPE_CHECKING
 
 from aiohttp import FormData
 
-from .invite import Invite
 from .channel import Channel
-from ..message.emoji import Emoji
+from .invite import Invite
+from .scheduled_events import ScheduledEvent
```
Comment (trag1c) [inline pincer/objects/guild/guild.py:14]: Move to ~L32

File: pincer/objects/guild/scheduled_events.py [hunk 8]
```diff
@@ -0,0 +1,107 @@
+from dataclasses import dataclass
+from enum import IntEnum, Enum
+from typing import Optional
+
+from ..guild.stage import PrivacyLevel
+from ..user import User
+from ...utils.api_object import APIObject, MISSING
+from ...utils.snowflake import Snowflake
+from ...utils.timestamp import Timestamp
+from ...utils.types import APINullable
```
Comment (trag1c) [inline pincer/objects/guild/scheduled_events.py:10]:
```suggestion
from typing import Optional, TYPE_CHECKING
from ...utils.api_object import APIObject, MISSING

if TYPE_CHECKING:
    from ..guild.stage import PrivacyLevel
    from ..user import User
    from ...utils.snowflake import Snowflake
    from ...utils.timestamp import Timestamp
    from ...utils.types import APINullable
```

**Review by trag1c — CHANGES_REQUESTED** [review]

**Review by trag1c — APPROVED** [review]

**Pull request merged**